kornia/augmentation/container/patch.py

```diff
@@ -0,0 +1,267 @@
+    afterwards. Different image processing and augmentation methods will be performed on each patch region.
+
+    Args:
+        *args (nn.Module): a list of processing modules.
+        same_on_batch (bool, optional): apply the same transformation across the batch.
+            If None, it will not overwrite the function-wise settings. Default: None.
+        keepdim (bool, optional): whether to keep the output shape the same as input (True) or broadcast it
+            to the batch form (False). If None, it will not overwrite the function-wise settings. Default: None.
+        patchwise_apply (bool, optional): apply image processing args will be applied patch-wisely.
+            if ``True``, the number of args must be equal to grid number.
+            if ``False``, the image processing args will be applied as a whole to all patches. Default: False.
+
+    Return:
+        List[Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]]: the tensor (, and the transformation matrix)
+            has been sequentially modified by the args.
+
+    Examples:
+        >>> import kornia.augmentation as K
+        >>> input = torch.randn(2, 3, 224, 224)
+        >>> seq = PatchSequential(
+        ...     ImageSequential(
+        ...         K.ColorJitter(0.1, 0.1, 0.1, 0.1, p=0.5),
+        ...         K.RandomPerspective(0.2, p=0.5),
+        ...         K.RandomSolarize(0.1, 0.1, p=0.5),
+        ...     ),
+        ...     K.RandomAffine(360, p=1.0),
+        ...     ImageSequential(
+        ...         K.ColorJitter(0.1, 0.1, 0.1, 0.1, p=0.5),
+        ...         K.RandomPerspective(0.2, p=0.5),
+        ...         K.RandomSolarize(0.1, 0.1, p=0.5),
+        ...     ),
+        ...     K.RandomSolarize(0.1, 0.1, p=0.1),
+        ... grid_size=(2,2),
+        ... patchwise_apply=False,
+        ... same_on_batch=True,
+        ... )
+        >>> seq(input).shape
+        torch.Size([2, 3, 224, 224])
+    """
+
+    def __init__(
+        self,
+        *args: nn.Module,
+        grid_size: Tuple[int, int] = (4, 4),
+        padding: str = "same",
+        same_on_batch: Optional[bool] = None,
+        keepdim: Optional[bool] = None,
+        patchwise_apply: bool = False,
+    ) -> None:
+        super(PatchSequential, self).__init__(
+            *args, same_on_batch=same_on_batch, return_transform=False, keepdim=keepdim
+        )
+        assert padding in ["same", "valid"], f"`padding` must be either `same` or `valid`. Got {padding}."
+        self.grid_size = grid_size
+        self.padding = padding
+        self.patchwise_apply = patchwise_apply
+        if patchwise_apply:
+            assert len(args) == grid_size[0] * grid_size[1], (
+                "The number of processing modules must be equal with grid size."
+                f"Got {len(args)} and {grid_size[0] * grid_size[1]}."
+            )
+
+    def is_intensity_only(self) -> bool:
```

Comment:
what's the need for this?

Reply:
When combine with ```AugmentationSequential```, it will be used to check if the ```return_transform``` is safe.